solutions/bio1/StringRewriting/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { SCodeBlock, SList, SText, STitle } from "components";
+
+export const StringRewriting = () => {
+  return (
+    <>
+      <STitle>Idea</STitle>
+      <SText>
+        Note $2^31$ is much greater than $10^9$, therefore a bruteforce approach won't work. 
+        Let $A[i], B[i], C[i], D[i], E[i]$ each respectively denote the result of $i$ operations applied to $A$, $B$, $C$, $D$ and $E$.  
+      </SText>
+      <SText>
+        From caseworking by hand, the following could be observed: 
+      </SText>
+      <SList>
+        <SText>$A[i] = A[i-1] + A[i-2]$</SText>
+        <SText>$B[i] = A[i+1]$</SText>
+        <SText>$C[i] = C[i-1] + D[i-1]</SText>
+        <SText>$D[i] = D[i-1] + C[i-1]</SText>
+        <SText>$E[i] = E[i-1] + E[i-1]$</SText>
+      </SList>
+      <SText>
+        This means a divide-and-conquer approach could be used for this question. 
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        Initially, we precompute 2 sequences and store them into arrays: powers of 2 and the fibonacci sequence. This will allow $O(1)$ retrieval for further queries regarding these 2 sequences. 
+      </SText>
+      <SText>
+        We will store the answer in a vector with size 5 called $ans$, initially filled with 0s. 
+        We will also store the length has been processed so far in $len$. By keeping track of this variable, we can ensure the information contributing to $ans$ is only when $len <= pos$. 
```

Comment:
Need to use \le instead of <=

Reply:
Thanks so much!! 
@myst-6 @macaquedev The preview is deployed now.